engine/termops.mli

```diff
@@ -250,6 +250,7 @@ val map_named_decl : ('r1 -> 'r2 ) -> ('a -> 'b) -> ('a, 'a, 'r1) Context.Named.
 val clear_named_body : Id.t -> env -> env
 
 val global_vars_set : env -> Evd.evar_map -> constr -> Id.Set.t
+val global_vars_set_no_evar : env -> Evd.evar_map -> constr -> Id.Set.t
```

Comment:
I don't think that exposing such a hackish function is a good idea. We already have several notions of "the set of variables in a term", adding one additional kind is a combinatorial explosion waiting to trigger. This function should be defined locally in declare and duly explained in comments there.

Reply:
I would say more ad hoc than hackish, but move done.